addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -1029,6 +1045,20 @@ export class Rtc extends Record {
         }
     }
 
+    async _upgradeConnection() {
+        const channelId = this.state.channel?.id;
+        if (!channelId) {
+            return;
+        }
+        if (!this.selfSession?.persona.isInternalUser) {
+            return;
+        }
+        if (this.serverInfo) {
+            return;
+        }
+        await rpc("/mail/rtc/channel/upgrade", { channel_id: channelId }, { silent: true });
```

Comment:
All of them could also be in a single `if ()`, again the `upgrade_to_sfu` is aborted in all cases anyway.

What's the subtlety about `!this.selfSession?.persona.isInternalUser`? Unauthenticated livechat visitors may also have a need to upgrade to SFU. Reminder that livechat is not absolutely 2 users in a call: more operators can join the call.

Reply:
@alexkuhn When there are more than 2 people, the channel will swap to SFU like before as usual, the `upgrade_to_sfu` is only a way to force it (in case of connection issues),

It will only happen when there is only 2 people in a call, in which case whoever calls the upgrade, upgrades the whole channel (it will `hot_swap` all participants). So if there is 1 visitor and 1 operator and a connection issue between them, the operator's client will be the one to request the upgrade.

I check for `isInternalUser` because not only would it be quite rare to have a call between 2 external users, with a connection issue (that TURNs do not save), but also because if we support it we have to lower the route security (and make a cors livechat one).